payment/src/main/java/org/killbill/billing/payment/core/sm/control/OperationControlCallback.java

```diff
@@ -99,16 +100,14 @@ public OperationResult doOperationCallback() throws OperationException {
                                                                                                      paymentStateControlContext.isApiPayment(),
                                                                                                      paymentStateContext.getCallContext());
 
-                final PriorPaymentControlResult pluginResult;
                 try {
-                    pluginResult = executePluginPriorCalls(paymentStateControlContext.getPaymentControlPluginNames(), paymentControlContext);
-                    if (pluginResult != null && pluginResult.isAborted()) {
-                        // Transition to ABORTED
-                        return PluginDispatcher.createPluginDispatcherReturnType(OperationResult.EXCEPTION);
-                    }
+                    executePluginPriorCalls(paymentStateControlContext.getPaymentControlPluginNames(), paymentControlContext);
                 } catch (final PaymentControlApiException e) {
                     // Transition to ABORTED and throw PaymentControlApiException to caller.
                     throw new OperationException(e, OperationResult.EXCEPTION);
+                } catch (final PaymentApiException paymentAbortedException) {
+                    // Transition to ABORTED
+                    throw new OperationException(paymentAbortedException, OperationResult.EXCEPTION);
```

Comment:
Yeah, my first implementation was based on what you first described, throwing an `OperationException` wrapping a `PaymentApiException`, but then we'd loose the plugin name that aborted the operation.

I agree that it can get kind of confusing, to only throw this exception for this particular case, but I'm not sure what's the best option.